rust/sedona-spatial-join/src/index.rs

```diff
@@ -557,32 +537,15 @@ impl SpatialIndex {
             }
         };
 
-        // Use pre-computed cached geometries for performance
-        let geometries = &self.cached_geometries;
-
-        if geometries.is_empty() {
-            return Ok(JoinResultMetrics {
-                count: 0,
-                candidate_count: 0,
-            });
-        }
-
-        // Choose distance metric based on use_spheroid parameter
-        let distance_metric: Box<dyn DistanceMetric<f32>> = if use_spheroid {
-            // For spheroid (geodesic) distance, we use the Haversine formula as an approximation for now.
-            // The distance metric will be used to calculate distances between geometries for ranking purposes.
-            Box::new(HaversineDistance::default())
-        } else {
-            Box::new(EuclideanDistance)
-        };
+        // Create an indexed distance metric adapter for KNN queries with cached distance metrics
+        let indexed_metric = self.create_knn_adapter(use_spheroid);
```

Comment:
The geometry cache in the KNN adapter is not reused across multiple `query_knn` calls. This renders the cache mostly useless.